xbmc/settings/AdvancedSettings.cpp

```diff
@@ -1385,16 +1387,27 @@ void CAdvancedSettings::AddSettingsFile(const std::string &filename)
   m_settingsFiles.push_back(filename);
 }
 
-float CAdvancedSettings::GetLatencyTweak(float refreshrate)
+float CAdvancedSettings::GetLatencyTweak(float refreshrate, bool isHDREnabled)
 {
   float delay = m_videoDefaultLatency;
-  for (int i = 0; i < (int) m_videoRefreshLatency.size(); i++)
+  if (isHDREnabled)
+    delay += m_videoDefaultHdrExtraLatency;
+
+  for (const auto& videolatency : m_videoRefreshLatency)
   {
-    RefreshVideoLatency& videolatency = m_videoRefreshLatency[i];
     if (refreshrate >= videolatency.refreshmin && refreshrate <= videolatency.refreshmax)
+    {
       delay = videolatency.delay;
-  }
 
+      if (isHDREnabled)
+      {
+        if (videolatency.hdrextradelay != 0.0f)
+          delay += videolatency.hdrextradelay;
+        else
+          delay += m_videoDefaultHdrExtraLatency;
```

Comment:
When HDR is enabled and `videolatency.hdrextradelay == 0,` `m_videoDefaultHdrExtraLatency` is added twice.

Global latency already added before for loop, then is not need this `else` code.

Reply:
Also the condition `!= 0.0f ` is redundant. For consistency with non-HDR delay can be simplified to:

```c++
if (isHDREnabled)
  delay += videolatency.hdrextradelay;
```

Reply:
It's not added twice because 'delay' is reinitialized with non-default value inside the loop. Keeps the default value only when no matching refresh node is found. 

The idea here is to re-apply  m_videoDefaultHdrExtraLatency in case user don't specify a value for 'hdrextradelay' under the refresh tag. IE : 
	<latency>
		<delay>-120</delay>
		<hdrextradelay>30</hdrextradelay>

		<refresh>
		  <min>23</min>
		  <max>25</max>
		  <delay>30</delay>
		</refresh>
	</latency>

But maybe that's not a good idea ?

Reply:
how about something like this (not runtime tested)
```suggestion
  float delay{};

  const auto& latency =
      std::find_if(m_videoRefreshLatency.cbegin(), m_videoRefreshLatency.cend(),
                   [refreshrate](const auto& param)
                   { return refreshrate >= param.refreshmin && refreshrate <= param.refreshmax; });

  if (latency != m_videoRefreshLatency.cend())
  {
    delay = latency->delay == 0.0 ? m_videoDefaultLatency : latency->delay;

    if (isHDREnabled)
    {
      if (latency->hdrExtraDelay == 0.0f)
        delay += m_videoDefaultHdrExtraLatency;
      else
        delay += latency->hdrExtraDelay;
    }
  }
  else
  {
    delay = isHDREnabled ? m_videoDefaultLatency
                         : m_videoDefaultLatency + m_videoDefaultHdrExtraLatency;
  }

```

Reply:
> how about something like this for clarity (not runtime tested)

Better yes, thank you. Also adds default 'delay' if not set in refresh node, for consistency.
line 1428 did the opposite of what's needed though.

Cleaned & tested : 

  float delay{};
  const auto& latency =
      std::find_if(m_videoRefreshLatency.cbegin(), m_videoRefreshLatency.cend(),
                   [refreshrate](const auto& param)
                   { return refreshrate >= param.refreshmin && refreshrate <= param.refreshmax; });

  if (latency != m_videoRefreshLatency.cend()) //refresh rate specific setting is found
  {
    delay = latency->delay == 0.0f ? m_videoDefaultLatency : latency->delay;
    if (isHDREnabled) 
        delay += latency->hdrextradelay == 0.0f ? m_videoDefaultHdrExtraLatency : latency->hdrextradelay;
  }
  else //apply default delay settings
  {
    delay = isHDREnabled ? m_videoDefaultLatency + m_videoDefaultHdrExtraLatency
                         : m_videoDefaultLatency;
  }
  return delay; // in milliseconds